src/test/regress/citus_tests/join_gen/generator/join_gen.py

```diff
@@ -0,0 +1,413 @@
+import random
+
+from generator.random_selections import (
```

Comment:
I think a high-level code-flow explanation would be very useful for this file. It is kind of hard to follow what is going on.

Reply:
added comment block at the top of grammar syntax. should we add more?